src/crawlee/proxy_configuration.py

```diff
@@ -124,19 +123,19 @@ async def new_proxy_info(
         if url is None:
             return None
 
-        # httpx.URL port field is None for default ports
-        default_ports = {'http': 80, 'https': 443}
-        port = url.port or default_ports.get(url.scheme)
-        if port is None:
+        if url.port is None:
             raise ValueError(f'Port is None for URL: {url}')
 
+        if url.host is None:
+            raise ValueError(f'Host is None for URL: {url}')
+
         info = ProxyInfo(
             url=str(url),
             scheme=url.scheme,
             hostname=url.host,
-            port=port,
-            username=url.username,
-            password=url.password,
+            port=url.port,
+            username=url.user if url.user else '',
+            password=url.password if url.password else '',
```

Comment:
not a request, but we can use this:

```suggestion
            username=url.user or '',
            password=url.password or '',
```

which is shorter